### PR TITLE
FINERACT-1483: fix Sonar Security Hotspots

### PR DESCRIPTION
**Pull request by taskain7**
## Description

Concatenated SQL queries have been changed to prepared statement.
SQL String literals have been checked with OWASP for MySQL and Postgresql escapeLiteral for PSQL.
Reviewed SQL queries have been tagged with `//NOSONAR`.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/account/service/PortfolioAccountReadPlatformServiceImpl.java [hunk 1]
```diff
@@ -119,25 +119,25 @@ public PortfolioAccountData retrieveOne(final Long accountId, final Integer acco
             case LOAN:
                 sql = "select " + this.loanAccountMapper.schema() + " where ";
                 if (portfolioAccountDTO.getClientId() != null) {
-                    sql += " la.client_id = ? and la.loan_status_id in (" + defaultAccountStatus.toString() + ") ";
+                    sql += " la.client_id = ? and la.loan_status_id in (" + defaultAccountStatus + ") ";
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/account/service/PortfolioAccountReadPlatformServiceImpl.java]: Don't we want to move the defaultAccountStatus into a parameter as well?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/account/service/PortfolioAccountReadPlatformServiceImpl.java [hunk 2]
```diff
@@ -119,25 +119,25 @@ public PortfolioAccountData retrieveOne(final Long accountId, final Integer acco
...
                     sqlParams.add(portfolioAccountDTO.getClientId());
                 } else {
-                    sql += " la.loan_status_id in (" + defaultAccountStatus.toString() + ") ";
+                    sql += " la.loan_status_id in (" + defaultAccountStatus + ") ";
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/account/service/PortfolioAccountReadPlatformServiceImpl.java]: Same here.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/account/service/PortfolioAccountReadPlatformServiceImpl.java [hunk 3]
```diff
@@ -119,25 +119,25 @@ public PortfolioAccountData retrieveOne(final Long accountId, final Integer acco
...
                 }
 
-                accounts = this.jdbcTemplate.query(sql, this.loanAccountMapper, sqlParams.toArray());
+                accounts = this.jdbcTemplate.query(sql, this.loanAccountMapper, sqlParams.toArray()); // NOSONAR
             break;
             case SAVINGS:
                 sql = "select " + this.savingsAccountMapper.schema() + " where ";
                 if (portfolioAccountDTO.getClientId() != null) {
-                    sql += " sa.client_id = ? and sa.status_enum in (" + defaultAccountStatus.toString() + ") ";
+                    sql += " sa.client_id = ? and sa.status_enum in (" + defaultAccountStatus + ") ";
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/account/service/PortfolioAccountReadPlatformServiceImpl.java]: same here

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/account/service/PortfolioAccountReadPlatformServiceImpl.java [hunk 4]
```diff
@@ -119,25 +119,25 @@ public PortfolioAccountData retrieveOne(final Long accountId, final Integer acco
...
                     sqlParams.add(portfolioAccountDTO.getClientId());
                 } else {
-                    sql += " sa.status_enum in (" + defaultAccountStatus.toString() + ") ";
+                    sql += " sa.status_enum in (" + defaultAccountStatus + ") ";
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/account/service/PortfolioAccountReadPlatformServiceImpl.java]: same here

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/address/service/AddressReadPlatformServiceImpl.java [hunk 5]
```diff
@@ -198,31 +198,31 @@ public AddressData mapRow(final ResultSet rs, @SuppressWarnings("unused") final
...
 
-        return this.jdbcTemplate.query(sql, rm, new Object[] { clientid, typeid });
+        return this.jdbcTemplate.query(sql, rm, new Object[] { clientid, typeid }); // NOSONAR
     }
 
     @Override
     public Collection<AddressData> retrieveAddressbyTypeAndStatus(final long clientid, final long typeid, final String status) {
         this.context.authenticatedUser();
-        Boolean temp = false;
+        boolean temp;
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/address/service/AddressReadPlatformServiceImpl.java]: Can you merge these 2 lines into 1? Makes more sense.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/address/service/AddressReadPlatformServiceImpl.java [hunk 6]
```diff
@@ -198,31 +198,31 @@ public AddressData mapRow(final ResultSet rs, @SuppressWarnings("unused") final
...
 
-        return this.jdbcTemplate.query(sql, rm, new Object[] { clientid, typeid, temp });
+        return this.jdbcTemplate.query(sql, rm, new Object[] { clientid, typeid, temp }); // NOSONAR
     }
 
     @Override
     public Collection<AddressData> retrieveAddressbyStatus(final long clientid, final String status) {
         this.context.authenticatedUser();
-        Boolean temp = false;
+        boolean temp;
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/address/service/AddressReadPlatformServiceImpl.java]: Same here.

**Review by galovics — CHANGES_REQUESTED** [review]
Minor changes, other than that looks good. Thanks.

**Review by galovics — APPROVED** [review]

**Comment by galovics** [comment]
Good job. I'll wait for the builds to finish then I'll merge it.

**Comment by vidakovic** [comment]
Give me a ping when you want these to appear downstream... looks like it should be synced sooner than later.

**Pull request merged**